compiler/src/iree/compiler/Codegen/Common/TileDispatchUsingInterface.cpp

```diff
@@ -194,12 +194,82 @@ static SmallVector<scf::ForOp> generateTileLoopNest(
   return loops;
 }
 
+/// Replace the `flow.dispatch.tensor.store` of the `untiledValue` with a tiled
+/// `flow.dispatch.tensor.store` that writes only a tile of the result at
+/// offsets given by `tiledOffsets` and sizes given by `tiledSizes`, using
+/// `tiledValue` as the source.
+static LogicalResult replaceWithTiledVersionOfStore(
+    RewriterBase &rewriter, OpResult untiledValue, OpResult tiledValue,
+    ArrayRef<OpFoldResult> tileOffsets, ArrayRef<OpFoldResult> tileSizes) {
+  SmallVector<IREE::Flow::DispatchTensorStoreOp> storeOps;
+  for (OpOperand &use : untiledValue.getUses()) {
+    auto storeOp = dyn_cast<IREE::Flow::DispatchTensorStoreOp>(use.getOwner());
+    if (storeOp && storeOp.getValue() == use.get()) {
+      storeOps.push_back(storeOp);
+    }
+  }
+  if (storeOps.empty()) return success();
+  if (storeOps.size() != 1) {
+    return rewriter.notifyMatchFailure(untiledValue.getOwner(),
+                                       "expected a single store for the op");
+  }
+  auto storeOp = storeOps.front();
+
+  SmallVector<OpFoldResult> tileStrides(tileOffsets.size(),
+                                        rewriter.getIndexAttr(1));
+  SmallVector<OpFoldResult> combinedOffsets, combinedSizes, combinedStrides;
+  if (failed(IREE::Flow::foldOffsetsSizesAndStrides(
+          rewriter, storeOp.getLoc(), storeOp.getMixedOffsets(),
+          storeOp.getMixedSizes(), storeOp.getMixedStrides(),
+          storeOp.getDroppedDims(), tileOffsets, tileSizes, tileStrides,
+          combinedOffsets, combinedSizes, combinedStrides))) {
+    return rewriter.notifyMatchFailure(
+        storeOp, "failed to create tiled flow.dispatch.tensor.store op");
+  }
+
+  rewriter.create<IREE::Flow::DispatchTensorStoreOp>(
+      storeOp.getLoc(), tiledValue, storeOp.getTarget(),
+      storeOp.getTargetDims(), combinedOffsets, combinedSizes, combinedStrides);
+  rewriter.eraseOp(storeOp);
+  return success();
+}
+
+/// Replace all `flow.dispatch.tensor.store` operations that use values produced
+/// by `untiledOp` as source, with tiled stores, with tiled values produced by
+/// `tiledOp`.
+static LogicalResult replaceWithTiledVersionOfStores(
```

Comment:
nit: Wondering if `replaceAllStoresWithTiledVersion` will be better distinguished from `replaceWithTiledVersionOfStore`